x-pack/plugin/monitoring/src/internalClusterTest/java/org/elasticsearch/xpack/monitoring/integration/MonitoringIT.java

```diff
@@ -364,6 +364,7 @@ public void enableMonitoring() throws Exception {
 
         final Settings settings = Settings.builder()
                 .put("xpack.monitoring.collection.enabled", true)
+                .put("xpack.monitoring.exporters._local.type", "local")
```

Comment:
This is necessary since this test (and some others here) set the type via [node level setting](https://github.com/elastic/elasticsearch/pull/57399/files#diff-bab5e7d9636db9cadcd30fa5c91d4fc9R80) but the settings validation does not validate across node level vs. dynamic settings (they get merged after independent validation). 

It also looks like we don't support validation across persistent vs. transient settings either, which makes any of these cross settings dependencies validation a bit terse. This translates to a real change in behavior w.r.t monitoring, but cross level validation is a bigger issue then monitoring. The safety here (imo) outweighs the convenience of cross type setting validation (and already exists for all HTTP type settings).